### PR TITLE
Memory leak fix in UIButton

### PR DESCRIPTION
**Pull request by amagams**

### CODE REVIEW

File: Frameworks/UIKit/UIButton.mm [hunk 1]
```diff
@@ -182,7 +182,7 @@ - (void)_UIButton_initInternal:(WXFrameworkElement*)xamlElement {
     // Set the XAML element's name so it's easily found in the VS live tree viewer
     [_xamlButton setName:[NSString stringWithUTF8String:object_getClassName(self)]];
 
-    __weak UIButton* weakSelf = self;
+    __block UIButton* weakSelf = self;
```
Comment (oliversa-msft) [inline Frameworks/UIKit/UIButton.mm]: UIButton is not ARC-enabled. File a bug so we can move it to ARC-enabled and then the "weak" decoration will work.

File: Frameworks/UIKit/UIButton.mm [hunk 2]
```diff
@@ -764,6 +762,7 @@ - (void)dealloc {
     }
 
     _xamlButton = nil;
+    [super dealloc];
```
Comment (oliversa-msft) [inline Frameworks/UIKit/UIButton.mm]: We can remove if ARC-enabled?

File: Frameworks/UIKit/UIButton.mm (hunk 1 above)
Comment (amagams) [inline Frameworks/UIKit/UIButton.mm]: bug filed.

File: Frameworks/UIKit.Xaml/Button.xaml.cpp [hunk 3]
```diff
@@ -161,7 +161,9 @@ UIKIT_XAML_EXPORT void XamlRemoveLayoutEvent(const ComPtr<IInspectable>& inspect
 // This will be used by autolayout to get the intrinsic content size when XAML has done calculating the elements final desired size.
 Windows::Foundation::Size Button::ArrangeOverride(Windows::Foundation::Size finalSize) {
     __super::ArrangeOverride(finalSize);
-    _layoutHook->Invoke(nullptr, nullptr);
+	if (_layoutHook) {
+		_layoutHook->Invoke(nullptr, nullptr);
+	}
```
Comment (jaredhms) [inline Frameworks/UIKit.Xaml/Button.xaml.cpp]: tabs

**Pull request merged**